apps/prairielearn/src/batched-migrations/20250214171608_course_instance_usages__external_grading_jobs__backfill.ts

```diff
@@ -0,0 +1,26 @@
+import { z } from 'zod';
+
+import { makeBatchedMigration } from '@prairielearn/migrations';
+import { loadSqlEquiv, queryAsync, queryRow } from '@prairielearn/postgres';
+
+const sql = loadSqlEquiv(import.meta.url);
+
+const CUTOFF_DATE = '2025-02-15T00:00:00Z';
+
+export default makeBatchedMigration({
+  async getParameters() {
+    // First delete usage data older than a hard-coded date
+    await queryAsync(sql.delete_old_usages, { CUTOFF_DATE });
+
+    // Only backfill from submissions that are older than the cutoff date
+    const max = await queryRow(
+      sql.select_bounds,
+      { CUTOFF_DATE },
+      z.bigint({ coerce: true }).nullable(),
+    );
+    return { min: 1n, max, batchSize: 100_000 };
```

Comment:
This takes about 45 seconds per batch in prod, so should complete in a bit under one day.